internal/storage/postgres/sensor/get_test.go

```diff
@@ -0,0 +1,389 @@
+	t.Run("should return sensors by status", func(t *testing.T) {
+		// given
+		ctx := context.Background()
+		suite.ResetDB(t)
+		suite.InsertSeed(t, "internal/storage/postgres/seed/test/sensor")
+		r := NewSensorRepository(suite.Store, defaultSensorMappers())
+
+		// when
+		got, err := r.GetSensorByStatus(ctx, &getTestSensors()[0].Status)
+
+		// then
+		assert.NoError(t, err)
+		assert.Len(t, got, 1)
+		assert.Equal(t, getTestSensors()[0].ID, got[0].ID)
+		assert.Equal(t, getTestSensors()[0].Status, got[0].Status)
+	})
+
+	t.Run("should return empty slice when no sensors match status", func(t *testing.T) {
+		// given
+		ctx := context.Background()
+		suite.ResetDB(t)
+		r := NewSensorRepository(suite.Store, defaultSensorMappers())
+
+		// when
+		status := entities.SensorStatus("offline")
+		got, err := r.GetSensorByStatus(ctx, &status)
+
+		// then
+		assert.NoError(t, err)
+		assert.Empty(t, got)
+	})
+
+	t.Run("should return error when status is nil", func(t *testing.T) {
+		// given
+		ctx := context.Background()
+		suite.ResetDB(t)
+		r := NewSensorRepository(suite.Store, defaultSensorMappers())
+
+		// when
+		got, err := r.GetSensorByStatus(ctx, nil)
+
+		// then
+		assert.Error(t, err)
+		assert.Nil(t, got)
+	})
+
+	t.Run("should return error when context is canceled", func(t *testing.T) {
+		// given
+		r := NewSensorRepository(suite.Store, defaultSensorMappers())
+		ctx, cancel := context.WithCancel(context.Background())
+		cancel()
+
+		// when
+		got, err := r.GetSensorByStatus(ctx, &getTestSensors()[0].Status)
+
+		// then
+		assert.Error(t, err)
+		assert.Nil(t, got)
+	})
+}
+
+func TestSensorRepository_GetSensorDataByID(t *testing.T) {
+	t.Run("should return sensor data for valid id", func(t *testing.T) {
+		// given
+		ctx := context.Background()
+		suite.ResetDB(t)
+		suite.InsertSeed(t, "internal/storage/postgres/seed/test/sensor")
+		r := NewSensorRepository(suite.Store, defaultSensorMappers())
+
+		// when
+		got, err := r.GetSensorDataByID(ctx, 1)
+
+		// then
+		assert.NoError(t, err)
+		assert.NotEmpty(t, got)
+		for _, data := range got {
+			assert.Equal(t, int32(1), data.ID)
+			assert.NotZero(t, data.CreatedAt)
+			assert.NotZero(t, data.UpdatedAt)
+		}
+	})
+
+	t.Run("should return empty slice when no data found", func(t *testing.T) {
+		// given
+		ctx := context.Background()
+		suite.ResetDB(t)
+		r := NewSensorRepository(suite.Store, defaultSensorMappers())
+
+		// when
+		got, err := r.GetSensorDataByID(ctx, 999)
+
+		// then
+		assert.NoError(t, err)
+		assert.Empty(t, got)
+	})
+
+	t.Run("should return error when context is canceled", func(t *testing.T) {
+		// given
+		r := NewSensorRepository(suite.Store, defaultSensorMappers())
+		ctx, cancel := context.WithCancel(context.Background())
+		cancel()
+
+		// when
+		got, err := r.GetSensorDataByID(ctx, 1)
+
+		// then
+		assert.Error(t, err)
+		assert.Nil(t, got)
+	})
+}
+
+func getTestSensors() []*entities.Sensor {
```

Comment:
Wouldnt it be better to have the "fake" sensor data in a seperate file like with "internal/service/domain/tree/utils.go" for example?